needy/cache.py

```diff
@@ -0,0 +1,29 @@
+
+class Cache:
+    def __init__(self, configuration={}):
+        self.__configuration = configuration
+
+    @staticmethod
+    def type():
+        '''type of cache'''
+        raise NotImplementedError('type')
+
+    def description(self):
+        '''uri, path, or other configuration description'''
+        raise NotImplementedError('description')
+
+    def configuration(self):
+        '''configuration dictionary to serialize to configuration files'''
+        return self.__configuration
+
+    def archive(self, source, key):
+        '''make files and folders at source retrievable with the given key'''
+        raise NotImplementedError('archive')
+
+    def unarchive(self, destination, key):
```

Comment:
I like the names, but question:

If we're only storing files, can you think of a problem that might be encountered if we didn't store folder metadata like permissions? It's definitely important to store things like executable permissions for files which is why I wanted to wrap it with tar, but I'm not so sure about how directories might be affected.